proof/crefine/RISCV64/Arch_C.thy

```diff
@@ -1595,30 +1661,40 @@ lemma decodeRISCVFrameInvocation_ccorres:
      apply (rule ccorres_rhs_assoc)+
      apply (ctac add: setThreadState_ccorres)
        apply csymbr
+       apply (rule ccorres_nondet_refinement)
+        apply (rule is_nondet_refinement_bindE)
+         apply (rule is_nondet_refinement_refl)
+        apply (simp split: if_split)
+        apply (rule conjI[rotated], rule impI, rule is_nondet_refinement_refl)
+        apply (rule impI)
+        apply (rule is_nondet_refinement_alternative1)
+       apply (clarsimp simp: liftE_bindE)
+       apply (rule ccorres_add_returnOk)
        apply (ctac(no_vcg) add: performPageGetAddress_ccorres)
-         apply (rule ccorres_alternative2)
          apply (rule ccorres_return_CE, simp+)[1]
-        apply (rule ccorres_inst[where P=\<top> and P'=UNIV], simp)
-       apply wp+
+        apply (rule ccorres_return_C_errorE, simp+)[1]
+       apply (wpsimp wp: sts_invs_minor' ct_in_state'_set)+
      apply (vcg exspec=setThreadState_modifies)
-    apply (rule ccorres_rhs_assoc)+
 
     \<comment> \<open>PageUnmap\<close>
     apply (simp add: returnOk_bind bindE_assoc
                      performRISCV64MMUInvocations)
+    apply (rule ccorres_rhs_assoc)+
     apply (ctac add: setThreadState_ccorres)
       apply (ctac(no_vcg) add: performPageInvocationUnmap_ccorres)
+        apply (rule ccorres_gen_asm)
+        apply (erule ssubst[OF if_P, where P="\<lambda>x. ccorres _ _ _ _ _ x _"])
         apply (rule ccorres_alternative2)
         apply (rule ccorres_return_CE, simp+)[1]
        apply (rule ccorres_inst[where P=\<top> and P'=UNIV], simp)
-      apply wp
+      apply (wpsimp simp: performPageInvocation_def)
      apply (wp sts_invs_minor')
     apply simp
     apply (vcg exspec=setThreadState_modifies)
 
    \<comment> \<open>PageMap\<close>
    supply Collect_const[simp del]
-   apply (rename_tac word rights pg_sz maptype mapdata buffera cap excaps cte
+   apply (rename_tac word rights pg_sz maptype mapdata calla buffera cap excaps cte
```

Comment:
When renaming, we should not propagate Isabelle-generated names like `calla`. If `call` is already taken, then `call'` is more traditional for the concrete side.